### PR TITLE
AO3-6311 Edited comment reply notification sent to creator of deleted comment

### PR DESCRIPTION
**Pull request by EchoEkhi**
# Pull Request Checklist

* [x] Have you read ["How to write the perfect pull request"](https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/)?
* [x] Have you read the [contributing guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
* [x] Have you added [tests for any changed functionality](https://github.com/otwcode/otwarchive/wiki/Automated-Testing)?
* [x] Have you added the [Jira](https://otwarchive.atlassian.net) issue number
  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
* [x] Do you have fewer than 5 pull requests already open? If not, please wait
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-6311

## Purpose

Do not send a edited comment reply notification to creator of deleted comment

## Testing Instructions

Follow instructions on JIRA.

## Credit

EchoEkhi (He/Him/{/([A-Z][a-z]*)/g})

### CODE REVIEW

File: features/comments_and_kudos/add_comment.feature [hunk 1]
```diff
@@ -117,6 +117,22 @@ Scenario: Comment threading, comment editing
...
+      And 1 email should be delivered to "User_A"
+    When I am logged in as "User_A"
+      And I view the work "Generic Work" with comments
+      And I delete the comment
+    When I am logged in as "User_B"
+      And I view the work "Generic Work" with comments
+      And I follow "Edit"
+      And I fill in "Comment" with "B's improved comment (edited)"
+      And I press "Update"
+    Then 1 email should be delivered to "User_A"
```
Comment (sarken) [inline features/comments_and_kudos/add_comment.feature:134]: Isn't the goal not to send User_A an email here?

Comment (EchoEkhi) [reply features/comments_and_kudos/add_comment.feature:134]: Ah, yes, but that 1 email is the "You have a reply!" email from line 125. It checks that there isn't a second email sent to A.
I should add a comment in the test to make this clear.

Comment (sarken) [reply features/comments_and_kudos/add_comment.feature:135]: Ah, okay! 

In that case, the best thing to do would be add `And all emails have been delivered` after line 125 to basically reset the count, and then the last step can be `Then 0 emails should be delivered to "User_A"`

Comment (EchoEkhi) [reply features/comments_and_kudos/add_comment.feature]: Oh that's neat - I've always wondered what that does before now!

**Review by sarken — APPROVED** [review]
Thank you!

**Pull request merged**